posts/2022-12-06-22.0.0.13-beta.adoc

```diff
@@ -0,0 +1,352 @@
+
+[source,java]
+----
+@WebServlet("/OidcAnnotatedServlet")
+@OpenIdAuthenticationMechanismDefinition(
+                providerURI = "https://localhost:8920/oidc/endpoint/OP",
+                clientId = "clientId",
+                clientSecret = "clientSecret",
+                redirectURI = "https://localhost:8940/MyApp/Callback",
+                claimsDefinition = @ClaimsDefinition(callerNameClaim = "sub", callerGroupsClaim = "groupIds"))
+@DeclareRoles("all")
+@ServletSecurity(@HttpConstraint(rolesAllowed = "all"))
+public class OidcAnnotatedServlet extends HttpServlet {
+…
+}
+----
+
+You can enable the Application Security 5.0 (Jakarta Security 3.0) feature by using the `appSecurity-5.0` feature in the server.xml file:
+   
+[source,xml]
+----
+    <featureManager>
+        <feature>appSecurity-5.0</feature>
+    </featureManager>
+----
+
+
+For more information about the `@OpenIdAuthenticationMechanismDefinition` annotation and its attributes, refer to the https://jakarta.ee/specifications/security/3.0/jakarta-security-spec-3.0.html#openid-connect-annotation[2.4.4. OpenID Connect Annotation] section of the Jakarta Security 3.0 specification. Please refer to https://javadoc.io/doc/jakarta.security.enterprise/jakarta.security.enterprise-api/latest/jakarta.security/module-summary.html[Jakarta Security Module] for the documentation of the Jakarta Security 3.0 APIs. 
+
+  
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC> 
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/23224
+// Contact/Reviewer: volosied,pnicolucci,ReeceNana
+// // // // // // // // 
+[#faces]
+== Jakarta Faces 4.0
+
+Jakarta Faces is a Model-View-Controller (MVC) framework for building web applications. It offers many convenient features, such as state management and input validation.
+
+Numerous changes have occurred since the last Jakarta Faces release. Notably, Pages (formerly JSP) support has been removed. Managed Beans support has also been removed, and developers should use CDI beans from this version forward. The namespaces also have been updated from \`http://xmlns.jcp.org/jsf/*` to \`jakarta.faces.*`. Otherwise, new attributes have been added to existing tags: `multiple` and `accept` for `h:inputFile` and `onerror` for `f:websocket`. Developers can now create facelets programmatically rather than building them via XML. Extensionless mapping is implemented, so URLs no longer need to include `/faces` or `.xhtml` mappings. Two new tags have been added to create a nested collections (i.e., optgroup containing option elements): `f:selectItemGroups` and `f:selectItemGroup`.  These are just some significant changes, but there are many more.

```

Comment:
```suggestion
Numerous changes have occurred since the last Jakarta Faces release. Notably, Pages (formerly JSP) support has been removed. Managed Beans support has also been removed, and developers should use CDI beans from this version forward. The namespaces also have been updated from `http://xmlns.jcp.org/jsf/*` to `jakarta.faces.*`. Otherwise, new attributes have been added to existing tags: `multiple` and `accept` for `h:inputFile` and `onerror` for `f:websocket`. Developers can now create facelets programmatically rather than building them via XML. Extensionless mapping is implemented, so URLs no longer need to include `/faces` or `.xhtml` mappings. Two new tags have been added to create a nested collections (i.e., optgroup containing option elements): `f:selectItemGroups` and `f:selectItemGroup`.  These are just some significant changes, but there are many more.

```